engine/src/main/java/org/destinationsol/game/Hero.java

```diff
@@ -0,0 +1,205 @@
+
+    public boolean isTranscendent() {
+        return isTranscendent;
+    }
+
+    public boolean isNonTranscendent() {
+        return !isTranscendent;
+    }
+
+    public Pilot getPilot() {
+        return isTranscendent ? transcendentHeroShip.getPilot() : hero.getPilot();
+    }
+
+    public SolShip getHero() {
+        if (isTranscendent) {
+            throw new SolDescriptiveException("Something is trying to get a SolShip hero while the hero is Transcendent state.");
+        }
+        return hero;
+    }
+
+    public SolShip getHeroUnchecked() {
+        return hero;
+    }
+
+    public StarPort.Transcendent getTranscendentHero() {
+        if (!isTranscendent) {
+            throw new SolDescriptiveException("Something is trying to get a Transcendent hero while the hero is in SolShip state.");
+        }
+        return transcendentHero;
+    }
+
+    public float getAngle() {
+        return isTranscendent ? transcendentHeroShip.getAngle() : hero.getAngle();
+    }
+
+    public Shield getShield() {
+        return isTranscendent ? transcendentHeroShip.getShield() : hero.getShield();
+    }
+
+    public Armor getArmor() {
+        return isTranscendent ? transcendentHeroShip.getArmor() : hero.getArmor();
+    }
+
+    public ShipAbility getAbility() {
+        onlySolShipHero();
+        return hero.getAbility();
+    }
+
+    public float getAbilityAwait() {
+        onlySolShipHero();
+        return hero.getAbilityAwait();
+    }
+
+    public boolean canUseAbility() {
+        onlySolShipHero();
+        return hero.canUseAbility();
+    }
+
+    public Vector2 getSpd() {
+        return isTranscendent ? transcendentHero.getSpd() : hero.getSpd();
+    }
+
+    public float getAcc() {
+        onlySolShipHero();
+        return hero.getAcc();
+    }
+
+    public Vector2 getPosition() {
+        return isTranscendent ? transcendentHero.getPosition() : hero.getPosition();
+    }
+
+    public Hull getHull() {
+        onlySolShipHero();
+        return hero.getHull();
+    }
+
+    public float getLife() {
+        return isTranscendent ? transcendentHeroShip.getLife() : hero.getLife();
+    }
+
+    public float getRotAcc() {
+        onlySolShipHero();
+        return hero.getRotAcc();
+    }
+
+    public float getRotSpd() {
+        onlySolShipHero();
+        return hero.getRotSpd();
+    }
+
+    public float getMoney() {
+        return isTranscendent ? transcendentHeroShip.getMoney() : hero.getMoney();
+    }
+
+    public void setMoney(float money) {
+        onlySolShipHero();
+        hero.setMoney(money);
+    }
+
+    public TradeContainer getTradeContainer() {
+        onlySolShipHero();
+        return hero.getTradeContainer();
+    }
+
+    public ItemContainer getItemContainer() {
+        return isTranscendent ? transcendentHeroShip.getIc() : hero.getItemContainer();
+    }
+
+    public void die() {
+        isDead = true;
+    }
+
+    public boolean maybeEquip(SolGame game, SolItem item, boolean equip) {
```

Comment:
I understand that the underlying method in `SolShip` is named the same, but since we are building a wrapper around it, consider renaming the method to follow the `tryXYZ` convention _if_ it is decided upon.

Reply:
In DS, convention is `maybe`. Also, this methods behaves in two ways depending on how the `equip` flag is set. If it is unset, it checks, whether the item can even be equipped. Otherwise, it tries to equip it and reports result

Reply:
The method name is perfectly fine for the existing DS codebase, true.

However, I would like to rename all "maybe" methods as well, since they make no sense to a newcomer.